tpchgen/src/distribution.rs

```diff
@@ -214,142 +214,148 @@ impl DistributionLoader {
 /// Distributions wraps all TPC-H distributions and provides methods to access them.
 #[derive(Debug, Clone)]
 pub struct Distributions {
-    distributions: IndexMap<String, Distribution>,
+    distributions: IndexMap<String, Arc<Distribution>>,
```

Comment:
The key design of this PR is to make multiple instances of `LineItemGeneratorIterator` which each generate data in parallel for different parts of the dataset

Initially, `clone`ing  `Distribution` consumed most of the time, and so in this PR I updated them to all be `Arc`d 

I think @scsmithr took a similar approach (using lifetimes) in this PR:
- https://github.com/clflushopt/tpchgen-rs/pull/35

Reply:
Rebased this PR to use that and it works great!